### PR TITLE
Update README with design notes

### PR DESCRIPTION
**Pull request by breville**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,6 +27,63 @@ The standalone runtime is also deployed to https://code-dot-org.github.io/ml-pla
...
+
+The creation of Dynamic Instructions was the breakthrough that gave us the best of both worlds.  They can be authored in levelbuilder, use the existing text-to-speech system, and can also be used outside of AI Lab.  They encourage instructions to be written for just the current panel on the screen.  (They can also be triggered by other states in the app for specific instructions at that moment in time).  They do not dynamically resize, so the screen below them does not shift around, and they are guaranteed to always be fully readable without needing scrolling.  And best of all, they empower curriculum designers to explain what's happening in each level with minimal changes required in the app itself.  (That said, showing new Dynamic Instructions for specific states in the app is very easy.)
+
```
Comment (Erin007) [inline README.md:39]: the teeniest of nits: do you want the . outside of the )?

File: README.md [hunk 2]
```diff
@@ -27,6 +27,63 @@ The standalone runtime is also deployed to https://code-dot-org.github.io/ml-pla
...
+#### Responsiveness
+The app is horizontally responsive, though it doesn't need to collapse all the way to mobile widths because our host site manages the viewport on mobile devices and also enforces landscape viewing.
+
+The app is vertically responsive, and uses flexbox to fill the vertical space with a combination of fixed- and variable-height elements.
+
+### Scenes:
+
+#### Select dataset
+This is usually the first scene, and can offer selection of both "pre-canned" datasets, which have accompanying metadata, or user-uploaded CSV.  The tiles use an art style somewhat consistent with that used elsewhere in our product.  There is a fun "grow" animation on tile hover, just to feel a little more interactive, which is reused for the A.I. bot head elsewhere in the app.
```
Comment (Erin007) [inline README.md:48]: I don't know how extensive we want this to be, but there might be documentation we could link to or reference here about how the Curriculum/Product team goes about adding a new dataset and accompanying image?

File: README.md [hunk 3]
```diff
@@ -27,6 +27,63 @@ The standalone runtime is also deployed to https://code-dot-org.github.io/ml-pla
...
+
+When selecting a label, the properties panel on the right shows information for the highlighted column:
+- a bar graph for categorical data;
+- min/max/range for numerical data.
+
+When selecting a feature, it shows the above properties for the highlighted column, but also attempts to show information about the relationship between that column and the previously-selected label:
+- a custom CrossTab when both label and feature are categorical data;
+- a scatterplot when both label and feature are numerical data.
+
+For student-uploaded CSV files, we attempt to guess the column type, but provide a way for the student to override this.
```
Comment (Erin007) [inline README.md:64]: maybe "we attempt to guess the column type based on the type of data detected in the column"?

File: README.md [hunk 4]
```diff
@@ -27,6 +27,63 @@ The standalone runtime is also deployed to https://code-dot-org.github.io/ml-pla
...
+
+#### Train Model
+A.I., the bot from AI for Oceans, is back.  Here, we attempt to illustrate the training process, in which a large amount of the original tabular data is given to the bot, row by row.  The animation is done by updating the React state 30 times a second to trigger a re-render each time.  The scene supports very small datasets, finishing early, as well as larger datasets, fading out after showing a few iterations of the animation.
+
+#### Generate Results
+In his scene we attempt to show A.I. generating results, again inspired by the way we presented it in AI for Oceans.  Here, the bot is making predictions on rows that were reserved for this purpose: it takes the features and predicts the label for each of those rows.  As in the previous scene, it handles both very small and larger datasets.
```
Comment (Erin007) [inline README.md:70]: typo: his -> this

File: README.md [hunk 5]
```diff
@@ -27,6 +27,63 @@ The standalone runtime is also deployed to https://code-dot-org.github.io/ml-pla
...
+- It shows the results of previous sets of predictions, with their respective statements.  This way the student can compare statements to see which have the highest "predictive power".
+- It lets the student view details of the most recent set of predictions, which shows in a pop-up.  This view has a toggle between showing correct and incorrect predictions.  In the table, the student can examine each row of reserved data to see how the label's actual value compares to what was predicted.
+- The student can "Try it out!" and run their own predictions.  The interface here is very similar to what they can see in App Lab once they import the saved model there.  The A.I. bot makes a reappearance here, because it's popular!
+
+#### Save Model
+
+The student can fill out the "model card" and then save the model, along with the model card information, to the server.  This model can then be imported in App Lab.  The model card information is seen in App Lab when previewing the model prior to importing it.
```
Comment (Erin007) [inline README.md:81]: I wonder about a sentence in here on the importance of Model Cards to the curriculum to justify why we included this step.

**Review by Erin007 — APPROVED** [review]
A few small thoughts on potential edits, but my overall feeling is "WOW! What a luxury for folks coming to this repo new".  Content like these few paragraphs aren't something I often see in our older READMEs and it feels like such a gift to our future selves. Thank you!

File: README.md (hunk 1 above)
Comment (breville) [reply README.md:39]: [No](https://blog.apastyle.org/apastyle/2013/03/punctuation-junction-periods-and-parentheses.html). :)

Comment (breville) [reply README.md:39]: Oh great catch; I got one right and one wrong! :|

File: README.md (hunk 2 above)
Comment (breville) [reply README.md:53]: Good call.  I added a small section about adding a new dataset.

**Pull request merged**